clients/drcachesim/tracer/output.cpp

```diff
@@ -989,9 +1002,26 @@ process_and_output_buffer(void *drcontext, bool skip_size_cap)
         if (op_offline.get_value() && op_split_windows.get_value())
             buf_ptr += instru->append_thread_exit(buf_ptr, dr_get_thread_id(drcontext));
     }
-
-    uintptr_t mode = tracing_mode.load(std::memory_order_acquire);
-
+    // Switch to instruction-tracing mode by adding FILTER_ENDPOINT marker if another
+    // thread triggered the switch.
+    ptr_int_t mode = tracing_mode.load(std::memory_order_acquire);
+    if (get_local_mode(data) != mode) {
+        if (get_local_mode(data) == BBDUP_MODE_L0_FILTER) {
+            NOTIFY(0, "Thread %d: filter mode changed\n", dr_get_thread_id(drcontext));
+            // If a switch occurred, then it is possible that the buffer
+            // contains a mix of filtered and unfiltered records. We discard
+            // the buffer by resetting the buffer pointer.
```

Comment:
This missing data might confuse some analyses and should be documented?

Reply:
This no longer applies since the data is not discarded anymore.